src/app/components/repositories/index.js

```diff
@@ -0,0 +1,153 @@
+              });
+          }));
+      });
+    }
+    return repositoriesLanguages;
+  }
+
+  filterReposByLanguage = (selectedList) => {
+    const { fetchedData: { repositories } } = this.props;
+    const filterLangArr = [];
+    const repoArr = repositories.nodes;
+    if (selectedList.length) {
+      for (let i = 0; i < selectedList.length; i++) {
+        filterLangArr.push(selectedList[i].name);
+      }
+      const filterRepoArr = repoArr.filter(repo => {
+        let repoLangArr = [];
+        repo.languages.nodes.map(lang => repoLangArr.push(lang.name));
+        const found = repoLangArr.some(r => filterLangArr.includes(r))
+        return found;
+      });
+      this.setState({
+        filterRepos: filterRepoArr,
+        selectedLanguages: filterLangArr,
+      });
+    } else {
+      this.setState({
+        filterRepos: repoArr,
+        selectedLanguages: [],
+      });
+    }
+  }
+
+  resetSelectedLanguages = () => {
+    this.multiselectRef.current.resetSelectedValues();
+  }
+
+  getMultiSelect = (repositories) => {
+    const uniqueLanguages = this.getLanguageSelector(repositories);
+    return (
+      repositories && (
+        <Multiselect
+          options={Object.values(uniqueLanguages)}
+          onSelect={this.filterReposByLanguage}
+          onRemove={this.filterReposByLanguage}
+          displayValue="name"
+          avoidHighlightFirstOption="false"
+          placeholder="Select Language"
+          ref={this.multiselectRef}
+        />
+      )
+    )
+  }
+
+  render() {
+    const { path, fetchedData: { repositories, repositories: { nodes } } } = this.props;
+    const { filterRepos, selectedLanguages } = this.state;
+    let viewedRepositories = filterRepos.length ? filterRepos : repositories.nodes;
+
+    return (
+      <div className="app">
+        <div className="d-flex">
+          <div className="form-left">
+            <p>
+              <strong>{REPO_COUNT_TEXT}{repositories.totalCount}</strong>
+            </p>
+            <p>Viewed Repositories: {viewedRepositories.length}</p>
+          </div>
+          <div className="form-right mt-20">
+            {
+              this.getMultiSelect(nodes)
+            }
+          </div>
+        </div>
+        <div className="repo">
+          <ul>
+            {viewedRepositories.map(repository => (
+              <React.Fragment key={repository.id}>
+                <li>
+                  <div className="mb-20">
+                    <NavLink to={`/${REPOSITORY_TEXT}/${path}/${repository.name}`}><strong>{repository.name}</strong></NavLink>
+                    {
+                      repository.isPrivate ? <span className="private-icon"><img src={LockIcon} alt="no-img" className="icon"></img>Private</span> : ''
+                    }
+                  </div>
+                  <div className="d-flex">
+                    <a className="repo-url" href={repository.url} target="_blank" rel="noopener noreferrer">{repository.url}</a>
```

Comment:
purpose of rel here?

Reply:
The rel attribute specifies the relationship between the current document and the linked document.